### ISSUE TITLE
Error with Gridap.jl + WriteVTK v1.7.1?

### ISSUE THREAD

**Issue by amartinhuertas** [gridap/Gridap.jl#286]
This is an error that I found while working with `GridapDistributed.jl` + `Gridap.jl` v0.10.4. Perhaps it is already known/solved. I document the error here for future reference, in a genuine application of Boy Scout's rule.

The following `GridapDistributed.jl` code:

```julia
function Gridap.writevtk(model::DistributedDiscreteModel,filebase::String)

  do_on_parts(model) do part, (model, gids)

    cdata = ["gids"=>gids.lid_to_gid,"owner"=>gids.lid_to_owner]
    filebase_part = filebase*"_$(part)"
    trian = Triangulation(model)
    writevtk(trian,filebase_part,celldata=cdata)
  end

end
```
crashes with:

```
CartesianDiscreteModels: Error During Test at /home/amartin/git-repos/GridapDistributed.jl/test/runtests.jl:11
  Got exception outside of a @test
  LoadError: MethodError: Cannot `convert` an object of type 
    WriteVTK.MeshCell{WriteVTK.VTKCellTypes.VTKCellType,Array{Int64,1}} to an object of type 
    WriteVTK.MeshCell{Array{Int64,1},V} where V<:(Union{AbstractArray{T,1}, Tuple{Vararg{T,N}} where N} where T<:Integer)
  Closest candidates are:
    convert(::Type{T}, !Matched::T) where T at essentials.jl:171
  Stacktrace:
   [1] push!(::Array{WriteVTK.MeshCell{Array{Int64,1},V} where V<:(Union{AbstractArray{T,1}, Tuple{Vararg{T,N}} where N} where T<:Integer),1}, ::WriteVTK.MeshCell{WriteVTK.VTKCellTypes.VTKCellType,Array{Int64,1}}) at ./array.jl:912
   [2] _generate_vtk_cells(::Gridap.Arrays.CachedArray{Int64,1,Array{Int64,1}}, ::Gridap.Arrays.Table{Int64,Int32}, ::Array{Int8,1}, ::Array{Int64,1}, ::Array{Array{Int64,1},1}) at /home/amartin/.julia/packages/Gridap/iXr2U/src/Visualization/Vtk.jl:184
   [3] _vtkcells at /home/amartin/.julia/packages/Gridap/iXr2U/src/Visualization/Vtk.jl:154 [inlined]
   [4] create_vtk_file(::Gridap.Visualization.VisualizationGrid{2,2,Float64}, ::String; celldata::Dict{Any,Any}, nodaldata::Dict{Any,Any}) at /home/amartin/.julia/packages/Gridap/iXr2U/src/Visualization/Vtk.jl:125
   [5] createvtk(::Gridap.Geometry.CartesianGrid{2,Float64,typeof(identity)}, ::String; order::Int64, nsubcells::Int64, celldata::Array{Pair{String,Array{Int64,1}},1}, cellfields::Dict{Any,Any}) at /home/amartin/.julia/packages/Gridap/iXr2U/src/Visualization/Vtk.jl:428
   [6] #writevtk#5 at /home/amartin/.julia/packages/Gridap/iXr2U/src/Visualization/Vtk.jl:395 [inlined]
   [7] (::GridapDistributed.var"#29#30"{String})(::Int64, ::Tuple{Gridap.Geometry.CartesianDiscreteModel{2,Float64,typeof(identity)},GridapDistributed.IndexSet{Array{Int64,1},Array{Int64,1},Dict{Int64,Int32}}}) at /home/amartin/git-repos/GridapDistributed.jl/src/DistributedDiscreteModels.jl:22
   [8] do_on_parts(::Function, ::GridapDistributed.SequentialCommunicator, ::GridapDistributed.DistributedDiscreteModel) at /home/amartin/git-repos/GridapDistributed.jl/src/SequentialCommunicators.jl:31
   [9] do_on_parts(::Function, ::GridapDistributed.DistributedDiscreteModel) at /home/amartin/git-repos/GridapDistributed.jl/src/Communicators.jl:17
   [10] writevtk(::GridapDistributed.DistributedDiscreteModel, ::String) at /home/amartin/git-repos/GridapDistributed.jl/src/DistributedDiscreteModels.jl:17
   [11] top-level scope at /home/amartin/git-repos/GridapDistributed.jl/test/CartesianDiscreteModelsTests.jl:69
   [12] include(::Module, ::String) at ./Base.jl:377
   [13] include at /home/amartin/git-repos/GridapDistributed.jl/test/runtests.jl:1 [inlined]
   [14] macro expansion at /home/amartin/git-repos/GridapDistributed.jl/test/runtests.jl:11 [inlined]
   [15] macro expansion at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.4/Test/src/Test.jl:1113 [inlined]
   [16] macro expansion at /home/amartin/git-repos/GridapDistributed.jl/test/runtests.jl:11 [inlined]
   [17] top-level scope at ./util.jl:175 [inlined]
   [18] top-level scope at /home/amartin/git-repos/GridapDistributed.jl/test/runtests.jl:0
   [19] include(::String) at ./client.jl:439
   [20] top-level scope at none:6
   [21] eval(::Module, ::Any) at ./boot.jl:331
   [22] exec_options(::Base.JLOptions) at ./client.jl:264
   [23] _start() at ./client.jl:484
  in expression starting at /home/amartin/git-repos/GridapDistributed.jl/test/CartesianDiscreteModelsTests.jl:69
```
This was caused by the following package update: `↑ WriteVTK v1.6.0 ⇒ v1.7.1`

**Comment by fverdugo** [comment]
I can confirm that a similar error has happened in Gridap#master on travis.

https://travis-ci.com/github/gridap/Gridap.jl/jobs/352008087

**Comment by amartinhuertas** [comment]
the definition of the struct causing trouble was eliminated from its original place

https://github.com/jipolanco/WriteVTK.jl/compare/v1.6.0...v1.7.0#diff-b7ef9ccb0f522307a0031c47082861b6L85

to

https://github.com/jipolanco/WriteVTK.jl/compare/v1.6.0...v1.7.0#diff-ed689fcc552afbea7351afb9d4ed2f0cR37

the definition of the struct seem to be quite different in v1.7.0

**Comment by fverdugo** [comment]
A temporary fix is on the way in PR #287

**Comment by amartinhuertas** [comment]
Solved in PR #324 ... closing the issue
